reporter/otlp_reporter.go

```diff
@@ -257,6 +274,11 @@ func setupGrpcConnection(parent context.Context, cfg *Config) (*grpc.ClientConn,
 			})))
 	}
 
+	if cfg.BasicAuth != "" {
```

Comment:
As @dmathieu wrote, given that this reporter will probably be dropped in the near future, have you considered implementing this functionality downstream in a separate reporter that you can keep maintaining (there is no long term maintenance guarantee for the `otlp_reporter`).

Additionally, this logic interacts with the `cfg.DisableTLS` option, e.g. if the latter is on then the following not particularly helpful failure message is logged: 

```
WARN[0000] Failed to setup gRPC connection (try 1 of 5): grpc: the credentials
 require transport level security (use grpc.WithTransportCredentials() to set) 
```

The code can be updated check if that's the case and abort with a more helpful error.